### PR TITLE
Support for simpleinit-msb init system

### PR DESCRIPTION
**Pull request by vaygr**
##### ISSUE TYPE

Feature Pull Request

##### COMPONENT NAME

`lib/ansible/modules/system/service.py `

##### ANSIBLE VERSION

`2.2.0.0`

##### SUMMARY

After [package management](https://github.com/ansible/ansible/pull/19072) it's the last of 2 main parts of any distro -- an init system.

This adds full support for `simpleinit-msb` used in [Source Mage](http://sourcemage.org).

[The init](http://sourcemage.org/Init) system is very simple and unique to Source Mage, based on the [Linux Boot Scripts](http://www.safe-mbox.com/~rgooch/linux/boot-scripts/) design pattern.

It uses `telinit` binary to manage init scripts in the system, move them between runlevels, enable/disable, and even install and remove them:

```text
telinit run <script> [<args>]        run init <script> with arguments <args>
telinit switch <rlvl>                switch to runlevel <rlvl>
telinit enable [<script> ... ]       enable and start <script> if not enabled
telinit disable [<script> ... ]      stop and disable <script> if not disabled
telinit bootenable [<script> ... ]   enable <script> if not enabled
telinit bootdisable [<script> ... ]  disable <script> if not disabled
telinit delete [<script> ... ]       delete <script> if disabled
telinit enabled                      list enabled startup scripts
telinit disabled                     list disabled startup scripts
telinit list                         list startup scripts and their runlevels
telinit runlevels                    list all runlevels
telinit runlevel                     print the highest completed runlevel
telinit move <script> <rlvl>         move <script> to runlevel <rlvl>
telinit install <file> <rlvl>        install <file> into runlevel <rlvl>
telinit spellinstall <spell>         install <spell>'s init or xinetd scripts
```

I've also managed to add some protection for non-existing scripts inside Ansible code.

### CODE REVIEW

**Comment by bcoca** [comment]
so the service module is being broken up into component modules in hopes of deprecating it soon, so I am remiss on adding this to it.

I would resubmit as it's own module and also add the init system to fact detection to set the ansible_service_mgr variable.

**Comment by vaygr** [comment]
I see the point. I just thought it might be less work for initial support, since it's not really that complicated change and most of patterns and code are re-used within the `service` module. Also it'd be great to see it coming along with `sorcery` module in 2.3 release.

File: lib/ansible/modules/system/service.py [hunk 1]
```diff
@@ -439,7 +439,7 @@ class LinuxService(Service):
     def get_service_tools(self):
 
         paths = [ '/sbin', '/usr/sbin', '/bin', '/usr/bin' ]
-        binaries = [ 'service', 'chkconfig', 'update-rc.d', 'rc-service', 'rc-update', 'initctl', 'systemctl', 'start', 'stop', 'restart', 'insserv' ]
+        binaries = [ 'service', 'chkconfig', 'update-rc.d', 'rc-service', 'rc-update', 'initctl', 'telinit', 'systemctl', 'start', 'stop', 'restart', 'insserv' ]
```
Comment (mattclay) [inline lib/ansible/modules/system/service.py]: CI failure due to PEP 8 issue:

```
2017-01-31 19:29:56 ERROR: PEP 8: lib/ansible/modules/system/service.py:442:161: E501 line too long (161 > 160 characters) (current)
```

**Comment by vaygr** [comment]
@mattclay thanks, corrected.

**Comment by vaygr** [comment]
@bcoca @mattclay is there a chance to ship it with upcoming **2.3**?

**Comment by gundalow** [comment]
CI was failing due to infrastructure issues bring up MacOS images (unrelated to this change) I've rescheduled the CI

**Comment by gundalow** [comment]
@vaygr Feature freeze for Ansible 2.3 was a number of weeks ago

**Comment by vaygr** [comment]
Oh, I'm too late to remind then. Hopefully it'll get its way into **2.4** with above mentioned refactoring or without, because init system support is quite important for full system support.

**Comment by vaygr** [comment]
Bump for updates :)

**Comment by vaygr** [comment]
Bumping for inclusion in 2.4!

**Comment by bcoca** [comment]
@vaygr my initial comment still stands, we are not adding functionality to the existing service module, so this should be submitted as a separate one.

**Comment by vaygr** [comment]
@bcoca oh, it wasn't not clear in your first comment that it's a **strict requirement** now. I'll see what I can do.

**Comment by vaygr** [comment]
@bcoca oh, it wasn't clear in your first comment that it's a **strict requirement** now. I'll see what I can do.

**Comment by gundalow** [comment]
I'm guessing this should be closed and a new PR raised for the new format?

needs_info

**Pull request closed without merging**